### PR TITLE
Platform refactored

### PR DESCRIPTION
**Pull request by aliartiza75**

### CODE REVIEW

File: configs/jenkins-maven-config.xml [hunk 1]
```diff
@@ -0,0 +1,75 @@
+<settings>
+    <!-- sets the local maven repository outside of the ~/.m2 folder for easier mounting of secrets and repo -->
+    <localRepository>${user.home}/.mvnrepository</localRepository>
+    <mirrors>
+        <mirror>
+            <id>nexus</id>
+            <mirrorOf>external:*</mirrorOf>
+            <url>http://nexus.release/repository/public/</url>
```
Comment (aliartiza75) [inline configs/jenkins-maven-config.xml:8]: I think this needs to be changed to `nexus.delivery` or `nexus-delivery`

File: configs/jenkins-maven-config.xml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        </mirror>
+    </mirrors>
+
+    <!-- lets disable the download progress indicator that fills up logs -->
+    <interactiveMode>false</interactiveMode>
+
+    <servers>
+        <server>
+            <id>docker-release.DOMAIN:443</id>
```
Comment (aliartiza75) [inline configs/jenkins-maven-config.xml]: same as above

File: scripts/configure-dev.sh [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+replace_values SLACK_INFRA_ALERTS_CHANNEL $SLACK_INFRA_ALERTS_CHANNEL && \
+replace_values SLACK_APPS_ALERTS_WEBHOOK_URL $SLACK_APPS_ALERTS_WEBHOOK_URL && \
+replace_values SLACK_APPS_ALERTS_CHANNEL $SLACK_APPS_ALERTS_CHANNEL && \
+replace_values GRAFANA_USERNAME $GRAFANA_USERNAME && \
+replace_values GRAFANA_PASSWORD $GRAFANA_PASSWORD
+
+# Replace following Configs with their base64 encoded values in secrets in platform
+replace_configs  BASE64_ENCODED_ALERTMANAGER_CONFIG ../configs/alertmanager.yaml && \
+replace_configs  BASE64_ENCODED_IMC_CONFIG ../configs/imc.yaml && \
+replace_configs  BASE64_ENCODED_JENKINS_CFG ../configs/jenkins.json && \
```
Comment (aliartiza75) [inline scripts/configure-dev.sh]: Convention over configuration CFG should be changed to CONFIG

File: variables.config [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+### [Nexus]
+NEXUS_ADMIN_ACCOUNT_USER=user-admin
+NEXUS_ADMIN_ACCOUNT_PASSWORD=L8TcnrwMytCFpAFe
+NEXUS_CLUSTER_ACCOUNT_USER=cluster-admin
+NEXUS_CLUSTER_ACCOUNT_PASSWORD=mrzUGWrD9buDYhMF
+
+### [Keycloak]
+KEYCLOAK_CLIENT_ID=stakater-online-platform
+KEYCLOAK_CLIENT_SECRET=1ce66f91-2068-4f3d-9578-f03fa8259230
```
Comment (aliartiza75) [inline variables.config]: is this not required to be in base64 encoded format?

File: .gitlab-ci.yml [hunk 5]
```diff
@@ -77,7 +77,7 @@ execute:
...
...TER_PLATFORM_SSH_GIT_URL STAKATER_PLATFORM_BRANCH=$STAKATER_PLATFORM_BRANCH DOMAIN=$DOMAIN BASE64_ENCODED_AWS_ACCESS_KEY_ID=$BASE64_ENCODED_AWS_ACCESS_KEY_ID BASE64_ENCODED_AWS_SECRET_ACCESS_KEY=$BASE64_ENCODED_AWS_SECRET_ACCESS_KEY BASE64_ENCODED_IMC_CONFIG=$BASE64_ENCODED_IMC_CONFIG BASE64_ENCODED_JENKINS_CFG=$BASE64_ENCODED_JENKINS_CFG KEYCLOAK_CLIENT_ID=$KEYCLOAK_CLIENT_ID KEYCLOAK_CLIENT_SECRET=$KEYCLOAK_CLIENT_SECRET BASE64_ENCODED_GITHUB_TOKEN=$BASE64_ENCODED_GITHUB_TOKEN BASE64_ENCODED_GITLAB_TOKEN=$BASE64_ENCODED_GITLAB_TOKEN BASE64_ENCODED_BITBUCKET_TOKEN=$BASE64_ENCODED_BITBUCKET_TOKEN BASE64_ENCODED_ADMIN_ACCOUNT_JSON=$BASE64_ENCODED_ADMIN_ACCOUNT_JSON BASE64_ENCODED_CLUSTER_ACCOUNT_JSON=$BASE64_ENCODED_CLUSTER_ACCOUNT_JSON NEXUS_ADMIN_ACCOUNT_USERNAME=$NEXUS_ADMIN_ACCOUNT_USERNAME NEXUS_CLUSTER_ACCOUNT_USERNAME=$NEXUS_CLUSTER_ACCOUNT_USERNAME BASE64_ENCODED_ALERTMANAGER_CONFIG=$BASE64_ENCODED_ALERTMANAGER_CONFIG BASE64_ENCODED_KEYCLOAK_CONFIG=$BASE64_ENCODED_KEYCLOAK_CONFIG BASE64_ENCODED_PROXYINJECTOR_CONFIG=$BASE64_ENCODED_PROXYINJECTOR_CONFIG BASE64_ENCODED_SLACK_CHANNEL=$BASE64_ENCODED_SLACK_CHANNEL BASE64_ENCODED_SLACK_WEBHOOK_URL=$BASE64_ENCODED_SLACK_WEBHOOK_URL
```
Comment (aliartiza75) [inline .gitlab-ci.yml]: i think  these  `BASE64_ENCODED_JENKINS_CFG` should be converted to `BASE64_ENCODED_JENKINS_CONFIG`. I think we should follow these conventions everywhere

File: Makefile [hunk 6]
```diff
@@ -24,18 +24,20 @@ BASE64_ENCODED_PROXYINJECTOR_CONFIG ?= BASE64_ENCODED_PROXYINJECTOR_CONFIG
...
 
 deploy:
-	bash install.sh
+	bash scripts/install.sh $(CLOUD_PROVIDER)
 	sleep 480
-	bash verify.sh
+	bash scripts/verify.sh
+
+verify:
+	bash scripts/verify.sh
```
Comment (aliartiza75) [inline Makefile]: @aliartiza75 need to change this file

File: README.md [hunk 7]
```diff
@@ -1,16 +1,6 @@
...
-This section provides a bird eye view of StakaterPlatform.
-
-### Problem Statement
-
-How can a novice use can deploy tools that is required for the management of kubernetes cluster?
-
-### Solution
-
-StakaterPlatform provides out of the box stacks to control, monitor, log, trace and security for applications deployed on kubernetes using CI/CD pipelines.
+StakaterPlatform Supercharges kubernetes clusters to provides out of the box stacks to control, monitor, log, trace and security tools.
```
Comment (aliartiza75) [inline README.md]: i think here we have some grammatical mistake

File: README.md [hunk 8]
```diff
@@ -20,161 +10,74 @@ StakaterPlatform consist of 6 stacks
...
+## Basic Configuration
+| Variables | Description | Default |  
+|---|---|---|
+| CLOUD_PROVIDER | Name of the k8s cloud provider | `nil` <br> (`aws` \| `azure`) |
+| DNS_PROVIDER | Cloud DNS Provider | `aws` (Route53) |
+| EXTERNAL_DNS_AWS_ACCESS_KEY_ID | AWS Access Key Id | `nil` |
+| EXTERNAL_DNS_AWS_SECRET_ACCESS_KEY | AWS Secret Access Key | `nil` |
+| DOMAIN | Domain to expose StakaterPlatform | `nil` |
+| BASE64_ENCODED_SSL_CERTIFICATE_CA_CRT | Base64 encoded Intermediate Certificate value | `nil` |
+| BASE64_ENCODED_SSL_CERTIFICATE_TLS_CRT | Base64 encoded Server Certficate value |`nil` |
```
Comment (aliartiza75) [inline README.md]: certificate typo

File: docs/detailed-config.md [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+## Global Variables
+| Variables | Description | Default |  
+|---|---|---|
+| CLOUD_PROVIDER | Name of the k8s cloud provider | `nil` <br> (`aws` \| `azure`) |
+| DNS_PROVIDER | Cloud DNS Provider | `aws` (Route53) |
+| EXTERNAL_DNS_AWS_ACCESS_KEY_ID | AWS Access Key Id | `nil` |
+| EXTERNAL_DNS_AWS_SECRET_ACCESS_KEY | AWS Secret Access Key | `nil` |
+| DOMAIN | Domain to expose StakaterPlatform | `nil` |
+| BASE64_ENCODED_SSL_CERTIFICATE_CA_CRT | Base64 encoded Intermediate Certificate value | `nil` |
+| BASE64_ENCODED_SSL_CERTIFICATE_TLS_CRT | Base64 encoded Server Certficate value |`nil` |
```
Comment (aliartiza75) [inline docs/detailed-config.md]: typo here too

File: scripts/destroy.sh [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+
+# Remove any remaing platform resources
+kubectl delete -f platform/
+
+# Remove Tiller
+helm reset --force
+
+# Remove Helm Operator
+helm delete --purge helm-operator
+
```
Comment (aliartiza75) [inline scripts/destroy.sh:17]: kindly add a TODO regarding role and cluster roles removal

File: .gitlab-ci.yml (hunk 5 above)
Comment (ahmedwaleedmalik) [reply .gitlab-ci.yml]: Ignore `makefile-ci` for now. We'll merge all of this into `makefile` eventually to keep things consistent

File: Makefile (hunk 6 above)
Comment (ahmedwaleedmalik) [reply Makefile]: Same here

File: README.md (hunk 7 above)
Comment (ahmedwaleedmalik) [reply README.md]: Some issue in PR it's not showing correct changes in this file.

File: scripts/configure-dev.sh (hunk 3 above)
Comment (ahmedwaleedmalik) [reply scripts/configure-dev.sh]: Ignore makefile-ci for now. We'll merge all of this into makefile eventually to keep things consistent

**Review by ahmedwaleedmalik — APPROVED** [review]

**Pull request merged**